### PR TITLE
ACCUMULO-4110 clean up the WALSunnyDayIT

### PR DESCRIPTION
**Pull request by mjwall**
Trying to make this IT fail less often.  See https://issues.apache.org/jira/browse/ACCUMULO-4110 for discussion.

### CODE REVIEW

File: test/src/main/java/org/apache/accumulo/test/functional/WALSunnyDayIT.java [hunk 1]
```diff
@@ -232,4 +216,53 @@ private void writeSomeData(Connector conn, String tableName, int row, int col) t
...
+    int waitLonger = getWaitFactor();
+    for (int i = 1; i <= TIMES_TO_COUNT; i++) {
+      Thread.sleep(i * PAUSE_BETWEEN_COUNTS * waitLonger);
+      wals = _getWals(c);
+      if (wals.size() == expectedCount) {
+        return wals;
+      }
+    }
+
+    fail("Unable to get the correct number of WALs, expected " + expectedCount + " but got " + wals.toString());
```
Comment (joshelser) [inline test/src/main/java/org/apache/accumulo/test/functional/WALSunnyDayIT.java]: Would recommend removing this fail and let it bubble up to the caller. Let them decide what to do when they don't get the expected number of WALs.

**Comment by joshelser** [comment]
One minor thing to fix and then I think it's good. Thanks, Mike!

File: test/src/main/java/org/apache/accumulo/test/functional/WALSunnyDayIT.java (hunk 1 above)
Comment (mjwall) [inline test/src/main/java/org/apache/accumulo/test/functional/WALSunnyDayIT.java]: Can you clarify what you are asking here @joshelser?  I replaced existing asserts like the following

assertEquals(wals.toString(), 2, wals.size());

with this new method named getWALsAndAssertCount.  I still think it is an assert, just an assert that tries extra hard to pass.

Comment (joshelser) [inline test/src/main/java/org/apache/accumulo/test/functional/WALSunnyDayIT.java]: I am suggesting that you remove the line this comment is on (the `fail(..)`) and the return of an empty hashmap you have after that. If we exit the above for-loop, just `return wals;`.

Comment (mjwall) [inline test/src/main/java/org/apache/accumulo/test/functional/WALSunnyDayIT.java]: If I make that change, doesn't it change the assertions of the original test?  I started with what you suggest, without the for loop to retry, just to ensure I got the same behavior.  A run that previously failed on line 113 saying "should have 3 WALs after roll", started failing on line 115 saying "all WALs should be in use".    It didn't make sense to me, so I added back a fail statement if the correct count could not be collected.  Let's talk though this more tomorrow.  Thanks Josh

Comment (joshelser) [inline test/src/main/java/org/apache/accumulo/test/functional/WALSunnyDayIT.java]: Ahh, sorry, this is my bad. I didn't realize you also removed all of the `assertEquals()` calls in the test case. I saw https://github.com/apache/accumulo/pull/128/files/4cee43691dc191236682db1894854337812419bb#diff-e901412aabad86d17df0a2dc524dcde9R157 which confused me.

**Comment by mjwall** [comment]
Merged with https://github.com/apache/accumulo/commit/4adeb4bcbb0aa724cbd87e13ee60d27fcff877e4

**Pull request merged**